### ISSUE TITLE
bug: block doesn't work correctly 

### ISSUE THREAD

**Issue by gigasource** [aidenybai/million#444]
**Describe the bug**

**To Reproduce**
Cell
``` js
import {Item} from "@/stories/Cell/CellForm.tsx";
import {useState} from "react";
import {block} from "million/react";

interface CellProps {
	item: Item
}

export const Cell = block(({item}: CellProps) => {
	const [active, setActive] = useState<boolean>(false);
	function onItemChange() {
		setActive(true);
		console.log('onItemChange', item.name);
	}

	return <div className={`border-[1px] h-full ${active ? 'border-amber-500' : 'border-[#EEEEEE]'}`}
																onPointerDown={() => onItemChange()}>
		<p>Cell</p>
		<p>{item.name} {item.price}</p>
	</div>
})
```
Cell.stories.tsx
```js
import {Cell} from "./Cell.tsx";
import {Item} from "@/stories/Cell/CellForm.tsx";

export default {
	title: 'Example/Cell',
}

const item1 : Item = {name: 'A', price: 10};
const item2 : Item = {name: 'B', price: 10};
const item3 : Item = {name: 'C, price: 10};

export const CellEx0 = () => {
	return <div className={'flex h-[200px] gap-[10px]'}>
		<div className={'flex-1 flex-col flex gap-[10px]'}>
			<Cell item={item1}/>
			<Cell item={item2}/>
			<Cell item={item3}/>
		</div>
	</div>
}
```
When i click item A, then C , log show: onItemChange A
Without block then it works correctly

**Expected behavior**

**Screenshots**
<img width="1341" alt="image" src="https://github.com/aidenybai/million/assets/56248437/6afc7f89-6027-4053-83eb-8b930bea11b4">

**Device (please complete the following information):**
 - OS [e.g. iOS]

**Additional context**

**Comment by tobySolutions** [comment]
Hey @gigasource, are you by chance getting any errors or something?? 👀👀👀

**Comment by aidenybai** [comment]
Fixed in `v2.4.8-beta.2`!
